frontend/src/io-managers/input.ts

```diff
@@ -31,6 +31,13 @@ export function createInputManager(editor: Editor, container: HTMLElement, dialo
 
 	let viewportPointerInteractionOngoing = false;
 	let textInput = undefined as undefined | HTMLDivElement;
+	let inCanvas = true;
+
+	const app = window.document.getElementById("app");
+	app?.focus();
+	app?.addEventListener("blur", () => {
```

Comment:
We need to clean this up when `unbindListeners` is called. (This occurs every time hot module replacement updates the app during development, and my previous failure to clean things up caused breakage.) Are you able to integrate this event listener with the block of listeners declared above (`const listeners = ...`)?